src/core/node.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <memory>
+#include <functional>
+
+namespace mavsdk {
+
+class NodeImpl;
+class MavsdkImpl;
+class PluginImplBase;
+
+/**
+ * @brief This class represents a MAVLink node, which is EXACTLY 1 component on one system.
+ * In other words, this connects to anything which emits its own MAVLink heartbeat.
+ *
+ * Node objects are used to interact with different MAVLink enabled devices/software in
+ * a MAVLink network. They are not created directly by application code, but are returned
+ * by the Mavsdk class.
+ */
+class Node {
+public:
+    /** @private Constructor, used internally
+     *
+     * This constructor is not (and should not be) directly called by application code.
+     *
+     * @param parent `MavsdkImpl` dependency.
+     * @param system_id System id.
+     * @param comp_id Component id.
+     */
+    explicit Node(MavsdkImpl& parent, uint8_t system_id, uint8_t comp_id);
```

Comment:
I'm not completely clear about the `node_id` used elsewhere in the code. Why is there a concept of `node_id`, but this concept does not exist inside the `Node` class? :thinking:

Reply:
@JonasVautherin node_id is a single 16 bit integer, which is easier to manipulate than 2x8bit (sid & cid) since I just bitshift and bitwise OR it. It doesn't make sense to me that the user would ever want to use this, however, so I haven't exposed it.

Reply:
For instance in a map, it's easy to do map<uint16_t, whatever> as opposed to map<pair<uint8_t, uint8_t>, whatever>